File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    ans = 0
    
    def dfs(a):
        visited[a] = 1
        for j in range(n):
            if computers[a][j] and not visited[j]:
                dfs(j)
    
    visited = [0]*n
    for i in range(n):
        if not visited[i]:
            dfs(i)
            ans += 1  
    return ans

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let graph = Array.from({ length : n}, () => new Set());
    for(let i = 0; i < n; i++){
        for(let j = 0; j < n; j++){
            if(i != j && computers[i][j] === 1){
                graph[i].add(j);
                graph[j].add(i);
            }
        }
    }
    
    let ans = 0;
    let q = [];
    let visited = Array(n).fill(false);
    for(let i = 0; i < n; i++){
        if(!visited[i]){
            q.push(i);
            ans++;
            while(q.length > 0){
                let now = q.shift();
                for(const num of graph[now]){
                    if(!visited[num]){
                        q.push(num);
                        visited[num] = true;
                    }
                }
            }
        }
    }
    return ans;
}